Framework/HistogramData/src/HistogramIterator.cpp

```diff
@@ -0,0 +1,41 @@
+#include "MantidHistogramData/HistogramIterator.h"
+#include "MantidHistogramData/Histogram.h"
+#include "MantidHistogramData/HistogramItem.h"
+#include "MantidKernel/make_unique.h"
+
+using Mantid::HistogramData::Histogram;
+using Mantid::HistogramData::HistogramItem;
+using Mantid::HistogramData::HistogramIterator;
+using Mantid::Kernel::make_unique;
+
+void HistogramIterator::increment() {
+  if (m_index < m_histogram.size()) {
+    ++m_index;
+    m_currentItem = make_unique<HistogramItem>(m_histogram, m_index);
```

Comment:
`m_histogram` should not change, just the index so why create a new `HistogramItem` (also see note above on duplicate data in iterator)?